src/hypothesis/strategies.py

```diff
@@ -1287,6 +1361,27 @@ def __not_a_first_class_strategy(self, name):
             ) % (name,))
     return DataStrategy()
 
+
+def register_type_strategy(custom_type, strategy):
```

Comment:
How does this work with user defined generic types? (Asking both for myself and because the documentation should say so)